### DISCUSSION TITLE
Mantine Portal Performance

### DISCUSSION THREAD

**Discussion by tylerevans** [mantinedev/mantine#6209] [Q&A]
Hey guys, mantine is great.

I'm using mantine components in a form I've built.

I notice - if I have a number of select boxes, the dom is added by default to the body of the dom.
This grows alot as we have longer form:

What is mantine doing about performance of this?
What is best practice when rendering for example 10 select boxes in a form?

<img width="1048" alt="Screenshot 2024-05-09 at 12 46 08 PM" src="https://github.com/mantinedev/mantine/assets/6850548/f9393b1b-748c-474a-96fd-8d4a5f0fbd8c">

**Comment by rtivital** [comment]
1. What performance issues do you have?
2. If you have issues, why do you think they are caused by the Portal component?
3. If you do not want elements to be rendered in portal, you can disable this behavior via component props.

**Reply by tylerevans** [reply]
It's more about, if I have a form with 50 selects, will it render the opened dropdown dom  - eg 50 portals.
See above screenshot ^.
Shouldn't it not render the dropdown only when you've activated a select dropdown? eg, then it adds the dom for the options?

**Comment by tylerevans** [comment]
@rtivital  here is an example per comment above.
This could grow to 100's dom.
<img width="543" alt="Screenshot 2024-05-10 at 12 19 10 PM" src="https://github.com/mantinedev/mantine/assets/6850548/9e1b5005-e415-4252-9dc3-aac578dec85b">

**Reply by rtivital** [reply]
Yes, that is right. If you render 100 components, these components are added to the DOM.
